app-crypt/nitrocli/nitrocli-0.2.0.ebuild

```diff
@@ -0,0 +1,54 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+CRATES="
+argparse-0.2.2
+bitflags-1.0.4
+cc-1.0.25
+fuchsia-zircon-0.3.3
+fuchsia-zircon-sys-0.3.3
+libc-0.2.45
+nitrocli-0.2.0
+nitrokey-0.2.1
+nitrokey-sys-3.4.1
+rand-0.4.3
+winapi-0.3.6
+winapi-i686-pc-windows-gnu-0.4.0
+winapi-x86_64-pc-windows-gnu-0.4.0
+"
+
+inherit cargo
+
+DESCRIPTION="A command line application for interacting with Nitrokey devices"
+HOMEPAGE="https://github.com/d-e-s-o/nitrocli"
+SRC_URI="$(cargo_crate_uris ${CRATES})"
+
+LICENSE="GPL-3+"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
+
+BDEPEND="
+	>=dev-lang/rust-1.31.0
+"
+DEPEND="
+	dev-libs/hidapi
+"
+# We require gnupg for /usr/bin/gpg-connect-agent.
+RDEPEND="
+	${DEPEND}
+	app-crypt/gnupg
+"
+
+QA_FLAGS_IGNORED="/usr/bin/nitrocli"
+
+src_install() {
+	cargo install -j $(makeopts_jobs) --path=. --root="${D}/usr" $(usex debug --debug "") \
+		|| die "cargo install failed"
+	rm "${D}/usr/.crates.toml" || die "failed to remove .crates.toml"
+
+	einstalldocs
+	doman "${S}/doc/nitrocli.1"
+}
```

Comment:
you mention test, but you don't define `src_test` so it uses default one that just runs
`emake test` and using that Makefile in the tarball.
is it what you really  wanted?

test just hangs for me btw, probably because of very high makejobs ( I have -j64 in MAKEOPTS)

but with this it does not hang:

```bash
src_test() {
        emake -j1 test
}
```
```sh
>>> Source compiled.
>>> Test phase: app-crypt/nitrocli-0.2.0
make --jobs=64 --load-average=63 --output-sync=target --print-directory -j1 test 
make: Entering directory '/tmp/portage/app-crypt/nitrocli-0.2.0/work/nitrocli-0.2.0'
   Compiling libc v0.2.45
   Compiling cc v1.0.25
   Compiling argparse v0.2.2
   Compiling rand v0.4.3
   Compiling nitrokey-sys v3.4.1
   Compiling nitrokey v0.2.1
   Compiling nitrocli v0.2.0 (/tmp/portage/app-crypt/nitrocli-0.2.0/work/nitrocli-0.2.0)
    Finished dev [unoptimized + debuginfo] target(s) in 56.38s
     Running target/debug/deps/nitrocli-77330e8c74d3d4bb

running 8 tests
test commands::tests::prepare_secret_ascii ... ok
test commands::tests::prepare_secret_non_ascii ... ok
test pinentry::tests::parse_pinentry_passphrase_error ... ok
test pinentry::tests::parse_pinentry_passphrase_good ... ok
test pinentry::tests::parse_pinentry_passphrase_unexpected ... ok
test pinentry::tests::parse_pinentry_response_ok ... ok
test pinentry::tests::parse_pinentry_response_ok_no_newline ... ok
test pinentry::tests::parse_pinentry_response_unexpected ... ok

test result: ok. 8 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```


but,  it seems because it uses makefile it runs compile all over again because of missing `--release`

probably this one will work better:
```bash
src_test() {
        RUST_TEST_NOCAPTURE=1 RUST_TEST_THREADS=1 cargo test -j $(makeopts_jobs) \
                $(usex debug "" --release) -v || die "tests failed"
}
```

and yes, it just re-uses already compiled crates
```bash
>>> Compiling source in /tmp/portage/app-crypt/nitrocli-0.2.0/work/nitrocli-0.2.0 ...
   Compiling libc v0.2.45
   Compiling cc v1.0.25
   Compiling argparse v0.2.2
   Compiling rand v0.4.3
   Compiling nitrokey-sys v3.4.1
   Compiling nitrokey v0.2.1
   Compiling nitrocli v0.2.0 (/tmp/portage/app-crypt/nitrocli-0.2.0/work/nitrocli-0.2.0)
    Finished release [optimized] target(s) in 32.74s
>>> Source compiled.
>>> Test phase: app-crypt/nitrocli-0.2.0
       Fresh cc v1.0.25
       Fresh argparse v0.2.2
       Fresh libc v0.2.45
       Fresh rand v0.4.3
       Fresh nitrokey-sys v3.4.1
       Fresh nitrokey v0.2.1
   Compiling nitrocli v0.2.0 (/tmp/portage/app-crypt/nitrocli-0.2.0/work/nitrocli-0.2.0)
     Running `rustc --edition=2018 --crate-name nitrocli src/main.rs --color always --emit=dep-info,link -C opt-level=3 --test -C metadata=4cfe76eaf7fa425a -C extra-filename=-4cfe76eaf7fa425a --out-dir /tmp/portage/app-crypt/nitrocli-0.2.0/work/nitrocli-0.2.0/target/release/deps -L dependency=/tmp/portage/app-crypt/nitrocli-0.2.0/work/nitrocli-0.2.0/target/release/deps --extern argparse=/tmp/portage/app-crypt/nitrocli-0.2.0/work/nitrocli-0.2.0/target/release/deps/libargparse-fe4541f8e7ab1baa.rlib --extern nitrokey=/tmp/portage/app-crypt/nitrocli-0.2.0/work/nitrocli-0.2.0/target/release/deps/libnitrokey-d364cb4aedbb3ac8.rlib -L native=/tmp/portage/app-crypt/nitrocli-0.2.0/work/nitrocli-0.2.0/target/release/build/nitrokey-sys-2c610e3d16a8207c/out`
    Finished release [optimized] target(s) in 0.99s
     Running `/tmp/portage/app-crypt/nitrocli-0.2.0/work/nitrocli-0.2.0/target/release/deps/nitrocli-4cfe76eaf7fa425a`

running 8 tests
test commands::tests::prepare_secret_ascii ... ok
test commands::tests::prepare_secret_non_ascii ... ok
test pinentry::tests::parse_pinentry_passphrase_error ... ok
test pinentry::tests::parse_pinentry_passphrase_good ... ok
test pinentry::tests::parse_pinentry_passphrase_unexpected ... ok
test pinentry::tests::parse_pinentry_response_ok ... ok
test pinentry::tests::parse_pinentry_response_ok_no_newline ... ok
test pinentry::tests::parse_pinentry_response_unexpected ... ok

test result: ok. 8 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```

Reply:
Hm, no idea why it would hang. I can't see the connection to `MAKEOPTS`. I've adjusted the ebuild to use `cargo test` as you suggested (the `RUST_*` settings aren't really necessary anymore, they have already been removed upstream)